### ISSUE TITLE
Bug when using style-loader through node

### ISSUE THREAD

**Issue by christianalfoni** [webpack/style-loader#47]
Hi there,

We are using a node instance that runs webpack-dev-server with style-loader. I have narrowed down the issue which you can test on this test repo: https://github.com/christianalfoni/style-loader-bug. As stated in the README:
1. `npm install`
2. `node server`
3. Go to `localhost:3000`
4. The background color should be red
5. If it is white, do a file change in `app/main.js`, just a lineshift, save and refresh... it works
6. If it is red, please restart server until the problem occurs

When increasing number of chunks you will also start to get random errors in those chunks.

When I do not use style-loader but manually create style tags using the css loader it does not give this error.

**Comment by bebraw** [comment]
:+1:

**Comment by sokra** [comment]
omg... this is pretty weird... :-O

**Comment by sokra** [comment]
in `server/bundle.js` you are starting the compiler twice:

``` js
  var compiler = webpack(webpackConfig, function () {
    console.log('Project is ready!');
  }); // starts a compilation

  var bundler = new WebpackDevServer(compiler, {
    // ...
  }); // starts watching, which also starts a compilation
```

to fix it:


``` js
  var compiler = webpack(webpackConfig); // only create a compiler

  var bundler = new WebpackDevServer(compiler, {
    // ...
  }); // starts watching, which starts the compilation
```

**Comment by sokra** [comment]
I propably need to throw an error in this case.

**Comment by christianalfoni** [comment]
Hehe, thanks for taking a look @sokra and yeah and error there would be good. Though I think a more common pattern using the webpack instance would be:

```js
  var compiler = webpack(webpackConfig);
  compiler(function () {
   console.log('compiled');
  });
```

And:

```js
  var compiler = webpack(webpackConfig);
  var bundler = new WebpackDevServer(compiler, {
    // First compile done and starts watching
  });
```

But I suppose that is too big of a change, and maybe does not make sense related to the rest of the API. But I do seem to loose a hook to check when the initial bundle is ready? As the callback for "new WebpackDevServer" is run immediately? Maybe add a second callback here that runs when the initial compile is done?

**Comment by christianalfoni** [comment]
Ah, I can see there is a plugin mechanism?

```js
  var compiler = webpack(webpackConfig);
  compiler.plugin('done', function () {
    console.log('Done bundling');
  });
  var bundler = new WebpackDevServer(compiler, {}, function () {
    // Initial compile done
  });
  // Starts compilation
```

No problems using that?

**Comment by SpaceK33z** [comment]
Closing since it looks like it's been fixed.
